packages/@aws-cdk/aws-iam/lib/role.ts

```diff
@@ -503,6 +506,20 @@ function createAssumeRolePolicy(principal: IPrincipal, externalIds: string[]) {
   return actualDoc;
 }
 
+function validateRolePath(path?: string) {
+  if (path === undefined || Token.isUnresolved(path)) {
+    return;
+  }
+
+  const validRolePath = /^(\/|\/[\u0021-\u007F]+\/)$/;
```

Comment:
I'm hesitant to allow validation of a character that will throw an error. I've put a ticket in with the IAM team to double check this. Can you provide the specific validation error you get with `/\u007F/` to help me look into this?

Reply:
That's great. I cannot investigate further on my side. A simple example with CDK is below. I've confirmed the same behavior with python client (boto3). I believe that the issue exists on IAM side.

```ts
import * as cdk from 'aws-cdk-lib';
import { aws_iam as iam } from 'aws-cdk-lib';


export class Example1Stack extends cdk.Stack {
  constructor(scope: cdk.App, id: string, props: cdk.StackProps) {
    super(scope, id, props);

    const role = new iam.Role(this, "role", {
      assumedBy: new iam.ServicePrincipal("lambda.amazonaws.com"),
      path: "/\u007F/",
    })
  }
}
```

Error messages

```
The stack named Example1Stack failed to deploy: UPDATE_ROLLBACK_COMPLETE: The specified value for path is invalid. It must begin and end with / and contain only alphanumeric characters and/or / characters. (Service: AmazonIdentityManagement; Status Code: 400; Error Code: ValidationError; Request ID: 08690aa1-f05c-4830-964d-acd4b2a84532; Proxy: null)
```

As far as I checked, other control charcters (e.g. `/\u0x7E/`) work and other patterns with `\u0x007F` (e. g. `/aaaa\u0x007Faaaaa/`) fail.